var/spack/repos/builtin/packages/sprng/package.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Sprng(AutotoolsPackage):
+    """SPRNG: A Scalable Library For Pseudorandom Number Generation
+       Sprng is a distributed process-aware random number generator that
+       avoids correlations in random number sequences across processes.
+    """
+
+    maintainers = ['kayarre']
+
+    homepage = "http://www.sprng.org"
+    url      = "http://www.sprng.org/Version5.0/sprng5.tar.bz2"
+
+    version('5.0', sha256='9172a495472cc24893e7489ce9b5654300dc60cba4430e436ce50d28eb749a66')
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('fortran', default=False, description='Enable Fortran support')
+
+    depends_on('mpi+cxx', when='+mpi')
```

Comment:
I didn't catch this on a first pass, sorry:
```suggestion
    depends_on('mpi', when='+mpi')
```
At the moment virtual dependencies don't have variants, just versions.

Reply:
thank you. this library won't compile without the cxx bindings (anecdotely) for mpi, is there another way to enforce that?

Reply:
Not at the moment, but we are thinking of ways to have stronger contracts for virtual dependencies. The best way you can deal with this right now is to check some property (like the runtime library?) and raise an exception if the condition you check is not satisfied. This part is also worth a TODO to be revisited later.

Reply:
this is great input. can you think of a package that has this that I can use as a good reference example?

Reply:
`#lazy`
modified to use 
`variant('parallel', default=True, description='Enable MPI support')`

and
`depends_on('mpi+cxx', when='+parallel')`

Reply:
@kayarre This is good to go with me if we drop `mpi+cxx`

Reply:
I dropped it and added what I believe to be a check for the relevant libraries?
Not sure if that is the best approach.